description: 'char: xillybus: Fix use-after-free in xillyusb_open()'
references:
- https://lore.kernel.org/all/20221022175404.GA375335@ubuntu/
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-45888
- https://bugzilla.redhat.com/show_bug.cgi?id=2148523
- https://www.cve.org/CVERecord?id=CVE-2022-45888
comments:
  debian/bwh: Driver was added in 5.14.
  debian/carnil: |-
    Negligible security impact, would need physical access to
    "exploit"
  masami: "XILLYUSB driver was added by a53d120 (\"char: xillybus: Add driver\n for\
    \ XillyUSB (Xillybus variant for USB)\" in 5.14-rc1. \nSo, before 5.14 kernels\
    \ are not affected."
  ubuntu/cascardo: requires physical USB access to disconnect
reporters:
- Hyunwoo Kim
introduced-by:
  mainline: [a53d1202aef122894b6e46116a92174a9123db5d]
fixed-by:
  mainline: [282a4b71816b6076029017a7bab3a9dcee12a920]
  stable/6.1: [c2dcdaa27d0a00bd8ac3a415d7f4c811631d4f5c]
  ubuntu/focal: ['version:5.4.0-9.12']
ignore:
  cip/4.19: Not affected
  cip/4.19-rt: Not affected
  cip/4.4: Not affected
  cip/4.4-rt: Not affected
  cip/4.4-st: Not affected
  cip/5.10: Not affected
  cip/5.10-rt: Not affected
  stable/4.14: Not affected
  stable/4.19: Not affected
  stable/4.9: Not affected
  stable/5.10: Not affected
  stable/5.4: Not affected
  stable/6.1: non issue, if anyone has physical access to trigger this they can do
    more harm anyway
